extensions/gcm_message_handler.go

```diff
@@ -353,14 +378,53 @@ func (g *GCMMessageHandler) sendMessage(message interfaces.KafkaMessage) error {
 	}
 
 	statsReporterHandleNotificationSent(g.StatsReporters, message.Game, "gcm")
+
+	gcmResMutex.Lock()
 	g.sentMessages++
-	l.WithFields(log.Fields{
+	gcmResMutex.Unlock()
```

Comment:
nit: consider using [Atomic.AddInt64](https://pkg.go.dev/sync/atomic#AddInt64) instead of Mutex just to lock and add 1 to this value

Reply:
I don't think using `atomic` would work well here, because in the `LogStats` function it will read all the values I'm locking at once and then resetting them, which would need the lock anyway